package.json

```diff
@@ -30,11 +30,13 @@
   },
   "homepage": "https://github.com/asyncapi/go-template#readme",
   "dependencies": {
+    "@asyncapi/generator": "^1.8.0",
+    "@asyncapi/parser": "^1.6.0",
     "@asyncapi/generator-filters": "^2.0.0",
     "@asyncapi/generator-hooks": "^0.1.0",
     "@asyncapi/generator-react-sdk": "^0.2.11",
-    "@asyncapi/modelina": "^0.24.0",
-    "puppeteer": "^5.2.1"
+    "@asyncapi/modelina": "^0.31.0",
+    "@babel/core": "^7.13.0"
```

Comment:
I don't think we need it, for sure not in `dependencies` and not even sure about `devDependencies` as it is needed for code transpilation and we don't do it here 🤔

Reply:
`@babel/core` is used for transpilation in the `generator` itself. We can remove it.

Reply:
Yeah, i thought exactly the same and I removed it but whenever I remove it, calling `ag ...` complains about missing dep. 
Why?

Reply:
Do you use the latest generator?

Reply:
It does not fail when running with the HTML template. This is the error it throws when generating with this template:

```
❯ ag -V
1.8.14

❯ ag ~/example.yaml . -o /tmp/wee
Something went wrong:
Error: Cannot find module '@babel/core'
Require stack:
- /Users/smoya/dev/go-template/node_modules/@rollup/plugin-babel/dist/index.js
- /Users/smoya/dev/go-template/node_modules/@asyncapi/generator-react-sdk/lib/transpiler/transpiler.js
- /Users/smoya/dev/go-template/node_modules/@asyncapi/generator-react-sdk/lib/transpiler/index.js
- /Users/smoya/dev/go-template/node_modules/@asyncapi/generator-react-sdk/lib/index.js
- /Users/smoya/dev/go-template/__transpiled/go.mod.js
- /usr/local/lib/node_modules/@asyncapi/generator/node_modules/@asyncapi/generator-react-sdk/lib/renderer/template.js
- /usr/local/lib/node_modules/@asyncapi/generator/node_modules/@asyncapi/generator-react-sdk/lib/renderer/index.js
- /usr/local/lib/node_modules/@asyncapi/generator/node_modules/@asyncapi/generator-react-sdk/lib/index.js
- /usr/local/lib/node_modules/@asyncapi/generator/lib/renderer/react.js
- /usr/local/lib/node_modules/@asyncapi/generator/lib/generator.js
- /usr/local/lib/node_modules/@asyncapi/generator/cli.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:902:15)
    at Function.Module._load (internal/modules/cjs/loader.js:746:27)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/smoya/dev/go-template/node_modules/@rollup/plugin-babel/dist/index.js:5:13)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/smoya/dev/go-template/node_modules/@asyncapi/generator-react-sdk/src/transpiler/transpiler.ts:4:1)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
```

Reply:
Ok, I had this same error, but please update deps to the latest versions like below:

```
    "@asyncapi/generator": "^1.8.14",
    "@asyncapi/parser": "^1.10.0",
    "@asyncapi/generator-filters": "^2.1.0",
    "@asyncapi/generator-hooks": "^0.1.0",
    "@asyncapi/generator-react-sdk": "^0.2.16",
    "@asyncapi/modelina": "^0.31.3"
```

and then everything should work :)

Reply:
what is generator and parser for?

Reply:
> and then everything should work :)

Do you mean without `babel/core`? It does not :/

```json
"dependencies": {
    "@asyncapi/generator": "^1.8.14",
    "@asyncapi/parser": "^1.10.0",
    "@asyncapi/generator-filters": "^2.1.0",
    "@asyncapi/generator-hooks": "^0.1.0",
    "@asyncapi/generator-react-sdk": "^0.2.16",
    "@asyncapi/modelina": "^0.31.3"
  }
```

```
❯ ag ~/example.yaml . -o /tmp/wee
Something went wrong:
Error: Cannot find module '@babel/core'
...
```

> what is generator and parser for?

+1 to this question. If I install babel/core and remove those two deps, it works.

Reply:
I have only these deps:

```js
    "@asyncapi/generator-filters": "^2.0.0",
    "@asyncapi/generator-hooks": "^0.1.0",
    "@asyncapi/generator-react-sdk": "^0.2.11",
    "@asyncapi/modelina": "^0.31.0"
```

and it works 😅  so please remove parser and generator, because as I see it's not used anywhere in the code.

Reply:
It worked. I definitely know nothing about npm. I mean, nothing.
![](https://i.giphy.com/media/boqCITo4CpcRYJObHN/giphy.webp)